go/tao/tpm2_tao.go

```diff
@@ -648,23 +656,92 @@ func (tt *TPM2Tao) Unseal(sealed []byte) (data []byte, policy string, err error)
 }
 
 func (s *TPM2Tao) InitCounter(label string, c int64) error {
-	fmt.Printf("tpm2.InitCounter\n") // REMOVE
-	return errors.New("InitCounter for tpm2 not implemented")
+	fmt.Printf("TPM2Tao.InitCounter\n")
+	// TODO(jlm): Change this?
+	if uint32(s.nvHandle) != 0 {
+		return nil
+	}
+	// TODO: make this more general?
+	s.nvHandle = tpm2.Handle(1000)
+	s.authString = "01020304"
+
+	return tpm2.InitCounter(s.rw, s.nvHandle, s.authString)
 }
 
 func (s *TPM2Tao) GetCounter(label string) (int64, error) {
-	fmt.Printf("tpm2.GetCounter\n") // REMOVE
-	return int64(0), errors.New("GetCounter for tpm2 not implemented")
+	fmt.Printf("TPM2Tao.GetCounter\n")
+	_ = s.InitCounter(label, int64(0))
+	return tpm2.GetCounter(s.rw, s.nvHandle, s.authString)
 }
 
+// Note:  Tpm2 counters work differently from other counters.  On startup, you
+// can't read a counter value before you initialize it which you do by incrementing
+// it.  What we do is use the (countvalue+1)/2 as the counter in RollbackSeal and Unseal.
+// When you seal, // if the current counter is odd, you bump it twice and use the 
+// value (countvalue+1)/2 in the counter slot.  If the counter is even, you bump by 1.
```

Comment:
We're relying on the fact that the linux_host never calls InitCounter on its host tao directly?